runtime/compiler/trj9/runtime/RelocationRecord.hpp

```diff
@@ -156,15 +156,17 @@ class TR_RelocationRecord
    {
    public:
       TR_RelocationRecord() {}
-      TR_RelocationRecord(TR_RelocationTarget *reloTarget, TR_RelocationRecordBinaryTemplate *rec, TR_RelocationRuntime *reloRuntime) : _record(rec), _reloRuntime(reloRuntime)
-         { decode(reloTarget); }
 
       void * operator new (size_t s, TR_RelocationRecord *p)   { return p; }
 
       virtual void print(TR_RelocationRuntime *reloRuntime);
       virtual char *name() { return "TR_RelocationRecord"; }
 
-      void decode(TR_RelocationTarget *reloTarget);
+     
+      static TR_RelocationRecord *create(TR_RelocationRecord *storage, TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget, TR_RelocationRecordBinaryTemplate *recordPointer)
+         {
+         return decode(storage, reloRuntime, reloTarget, recordPointer);
```

Comment:
Why not just fold decode() into create and keep create() out of the header? Nothing else calls decode, I don't think, so the extra call doesn't seem valuable to me.